src/lem/README.md

```diff
@@ -0,0 +1,94 @@
+## Lurk Evaluation Model (LEM)
+
+The main purpose of LEM is to represent Lurk's step function *as data*, which can
+be interpreted or used to synthesize the circuit.
+
+### Pointer cycle
+
+The variables of a LEM are represented by *meta pointers*, which become bound to
+concrete pointers during interpretation. So a `MetaPtr` is a named reference to
+a `Ptr`.
+
+To synthesize the circuit, the `Ptr`s that come from interpretation are hashed
+and become `ZPtr`s, whose data can be allocated in the circuit to generate
+*allocated pointers* (`AllocatedPtr`).
```

Comment:
Let's remember that we will need to factor out any such allocations shared between reductions so any 'constants' allocated appear only once per circuit. It's also worth noting that we may be able to avoid some such allocations entirely. We should be careful not to converge on a design that forces us to allocate constants needlessly.